bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -173,12 +202,16 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
                     if (Try {
                       val futDone = Await.ready(fut, 30.seconds)
```

Comment:
The problem here is that if we allow things to continue BEFORE the `Future` completes, the `Future` acts independently of the thread, and thus the thread ends and immediately a new thread is started by the executor without waiting for the previous catalog lookup to be done. Thus, things run as if we have a massive number of threads scheduled and bombard the catalog server.

I'm not sure if there is a better way to do this, unfortunately, as this is certainly not elegant.
For a for each of the future, will that suspend execution until the `Future` completes?